addons/mail/static/src/components/discuss/discuss.js

```diff
@@ -268,6 +268,12 @@ class Discuss extends Component {
         }
         this.discuss.clearReplyingToMessage();
         this.discuss.update({ activeMobileNavbarTabId: ev.detail.tabId });
+        // After loading Discuss from an arbitrary tab other then 'mailbox',
+        // switching to 'mailbox' requires to also set its inner-tab ;
+        // by default the 'inbox'.
+        if (ev.detail.tabId === 'mailbox') {
+            this.env.messaging.inbox.open();
+        }
```

Comment:
Proper way to fix this is to put logic in `models/discuss.js:_computeThread()` to auto-set `inbox` when `activeMobileNavbarTabId` is `mailbox` and current thread is not a mailbox.